src/common/args.cpp

```diff
@@ -720,15 +722,19 @@ fs::path ArgsManager::GetConfigFilePath() const
 
 ChainType ArgsManager::GetChainType() const
 {
-    std::string chain_str = GetChainTypeString();
-    std::optional<ChainType> chain = ChainTypeFromString(chain_str);
-    if (!chain) {
-        throw std::runtime_error(strprintf("Unknown chain %s.", chain_str));
-    }
-    return *chain;
+    std::variant<ChainType, std::string> arg = GetChainArg();
+    if (auto* parsed = std::get_if<ChainType>(&arg)) return *parsed;
+    throw std::runtime_error(strprintf("Unknown chain %s.", std::get<std::string>(arg)));
 }
 
 std::string ArgsManager::GetChainTypeString() const
+{
+    auto arg = GetChainArg();
+    if (auto* parsed = std::get_if<ChainType>(&arg)) return ChainTypeToString(*parsed);
+    return std::get<std::string>(arg);
+}
+
+std::variant<ChainType, std::string> ArgsManager::GetChainArg() const
```

Comment:
In commit "refactor: Avoid needless chain type string conversions" (95744f9cf1f143b6449a5046a914557b3886e3a2)

It seems like this commit could be moved before the "refactor: Replace string chain name constants with ChainTypes" (ca13faf0cc8acbaf425ac69c783415465403b5a3)" commit, which would simplify that commit, and avoid for the `ArgsManager::GetChainType()` method to change after it is added. I think the only change you would need to make to the code in this commit to move it earlier would be to keep the `ArgsManager::GetChainTypeString()` method called `ArgsManager::GetChainName()`